packages/cli/src/commands/deploy/spec.ts

```diff
@@ -0,0 +1,89 @@
+import { Question, Arg, Option, ArgsAndOpts } from '../../register-command';
+import { MethodArg } from '../../prompts/prompt';
+
+export const name = 'deploy';
+export const description = 'deploy a contract instance';
+
+export const args: Arg[] = [
+  {
+    name: 'contract',
+    async prompt(): Promise<Question> {
+      const choices = await import('../../prompts/choices');
+
+      // TODO: Make this include contracts from _unlinked_ dependencies.
+      const contracts = choices.contracts('all');
+
+      return {
+        message: 'Pick a contract to deploy',
+        choices: contracts,
+        validate: value => contracts.includes(value),
+      };
+    },
+  },
+  {
+    name: 'arguments',
+    variadic: true,
+    async prompt(argsAndOpts: ArgsAndOpts): Promise<Question[]> {
+      const { fromContractFullName } = await import('../../utils/naming');
+      const { default: ContractManager } = await import('../../models/local/ContractManager');
+      const { argLabelWithIndex } = await import('../../prompts/prompt');
+      const { parseArg, getSampleInput } = await import('../../utils/input');
+
+      const contractName = argsAndOpts.contract as string;
+
+      const { package: packageName, contract: contractAlias } = fromContractFullName(contractName);
+      const contract = new ContractManager().getContractClass(packageName, contractAlias);
+      const constructorInputs: MethodArg[] = contract.schema.abi.find(f => f.type === 'constructor')?.inputs ?? [];
```

Comment:
Same here as above, I'd refactor this line out. I'd also consider a helper function that returns the contract object directly from the `fullContractName`. And I'd love to turn the `ContractManager` into a set of standalone functions - why everything needs to be an object?!